src/Domain/Handler/SubscriptionPurchasedEventHandler.ts

```diff
@@ -58,37 +61,42 @@ implements DomainEventHandlerInterface
     user: User,
     subscriptionName: SubscriptionName
   ): Promise<void> {
-    const roleName = this.subscriptionNameToRoleNameMap.get(subscriptionName)
+    const currentRoleName = (await user.roles)[0].name as RoleName
+    const newRoleName = this.subscriptionNameToRoleNameMap.get(subscriptionName)
 
-    if (roleName === undefined) {
+    if (newRoleName === undefined) {
       this.logger.warn(
         `Could not find role name for subscription name: ${subscriptionName}`
       )
       return
     }
 
-    const role = await this.roleRepository.findOneByName(roleName)
+    const role = await this.roleRepository.findOneByName(newRoleName)
 
     if (role === undefined) {
-      this.logger.warn(`Could not find role for role name: ${roleName}`)
+      this.logger.warn(`Could not find role for role name: ${newRoleName}`)
       return
     }
 
     user.roles = Promise.resolve([role])
     await this.userRepository.save(user)
+    await this.webSocketsService.sendUserRoleChangedEvent(
+      user,
+      currentRoleName,
+      newRoleName
+    )
   }
 
   private async createSubscription(
     subscriptionName: string,
     user: User,
     subscriptionExpiresAt: number,
-    timestamp: number,
   ): Promise<void> {
     const subscription = new UserSubscription()
     subscription.planName = subscriptionName
     subscription.user = Promise.resolve(user)
-    subscription.createdAt = timestamp
-    subscription.updatedAt = timestamp
+    subscription.createdAt = dayjs.utc().valueOf()
```

Comment:
hmm `createdAt` and `updatedAt` values should come from the event - why the chnage?

Reply:
Ah wasn't sure, I figured maybe we would want to differentiate the event timestamp from the actual update to the db. Will revert this change 🙂